packages/@aws-cdk/aws-servicecatalog/test/tag-option.test.ts

```diff
@@ -0,0 +1,126 @@
+import { Template } from '@aws-cdk/assertions';
+import * as cdk from '@aws-cdk/core';
+import * as servicecatalog from '../lib';
+
+describe('TagOptions', () => {
+  let app: cdk.App;
+  let stack: cdk.Stack;
+
+  beforeEach(() => {
+    app = new cdk.App();
+    stack = new cdk.Stack(app);
+  });
+
+  describe('creating tagOption(s)', () => {
+    test('default tagOptions creation', () => {
+      new servicecatalog.TagOptions(stack, 'TagOptions', {
+        tagOptions: {
+          key1: ['value1', 'value2'],
+          key2: ['value1', 'value2', 'value3'],
+        },
+      });
+
+      Template.fromStack(stack).resourceCountIs('AWS::ServiceCatalog::TagOption', 5);
+    }),
+
+    test('fails to create tag option with invalid minimum key length', () => {
+      expect(() => {
+        new servicecatalog.TagOptions(stack, 'TagOptions', {
+          tagOptions: {
+            '': ['value1', 'value2'],
+          },
+        });
+      }).toThrowError(/Invalid TagOption key for resource/);
+    }),
+
+    test('fails to create tag option with invalid maxium key length', () => {
+      expect(() => {
+        new servicecatalog.TagOptions(stack, 'TagOptions', {
+          tagOptions: {
+            ['longKey'.repeat(1000)]: ['value1', 'value2'],
+          },
+        });
+      }).toThrowError(/Invalid TagOption key for resource/);
+    }),
+
+    test('fails to create tag option with invalid value length', () => {
+      expect(() => {
+        new servicecatalog.TagOptions(stack, 'TagOptions', {
+          tagOptions: {
+            key: ['tagOptionValue'.repeat(1000)],
+          },
+        });
+      }).toThrowError(/Invalid TagOption value for resource/);
+    }),
+
+    test('associate tag options', () => {
+      const portfolio = new servicecatalog.Portfolio(stack, 'MyPortfolio', {
+        displayName: 'testPortfolio',
+        providerName: 'testProvider',
+      });
+
+      const tagOptions = new servicecatalog.TagOptions(stack, 'TagOptions', {
+        tagOptions: {
+          key1: ['value1', 'value2'],
+          key2: ['value1', 'value2', 'value3'],
+        },
+      });
+      portfolio.associateTagOptions(tagOptions);
+
+      Template.fromStack(stack).hasResource('AWS::ServiceCatalog::TagOption', 5);
+      Template.fromStack(stack).hasResource('AWS::ServiceCatalog::TagOptionAssociation', 5);
+    }),
+
+    test('create and associate tag options to different resources', () => {
+      const portfolio1 = new servicecatalog.Portfolio(stack, 'MyPortfolio1', {
+        displayName: 'testPortfolio1',
+        providerName: 'testProvider1',
+      });
+
+      const portfolio2 = new servicecatalog.Portfolio(stack, 'MyPortfolio2', {
+        displayName: 'testPortfolio2',
+        providerName: 'testProvider2',
+      });
+
+      const tagOptions = new servicecatalog.TagOptions(stack, 'TagOptions', {
+        tagOptions: {
+          key1: ['value1', 'value2'],
+          key2: ['value1', 'value2', 'value3'],
```

Comment:
What if I do:

```
          key2: ['value2', 'value2', 'value2'],
```

? Seems like we are missing tests for these cases.

Reply:
I've added a test `tag options with duplicate values are idempotent`